domain/machine/service/service_test.go

```diff
@@ -90,3 +90,66 @@ func (s *serviceSuite) TestGetLifeError(c *gc.C) {
 	c.Check(err, jc.ErrorIs, rErr)
 	c.Assert(err, gc.ErrorMatches, `getting life status for machine "666": boom`)
 }
+
+func (s *serviceSuite) TestListAllMachinesSuccess(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	s.state.EXPECT().ListAllMachines(gomock.Any()).Return([]string{"666"}, nil)
+
+	machines, err := NewService(s.state).ListAllMachines(context.Background())
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(machines, gc.DeepEquals, []string{"666"})
+}
+
+func (s *serviceSuite) TestListAllMachinesError(c *gc.C) {
```

Comment:
I think this test should have some documentation to describe what it is doing. Specifically the docs should be saying we are trying to test that when we receive an error from the state layer it is properly maintained back up via the services layer.